### PR TITLE
[runtime] Add Dedup Support to MkBundle

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by alexanderkyte** [comment]
There's one managed test failing, MonoTests.Remoting.Http.Bug315570.Main, and it's because the fixture setup failed. This test passes on x86. Since the Linux branch is green on x64 but not on i386, can I consider this a flake? 

@akoeplinger @alexischr ?

**Comment by akoeplinger** [comment]
> can I consider this a flake?

@alexanderkyte yes

**Comment by luhenry** [comment]
@alexanderkyte the windows failure seems related to your changes as the test fails on both windows, and even after restarting the build it still fails.

**Comment by alexanderkyte** [comment]
build pkg

**Comment by alexanderkyte** [comment]
I've been investigating that

**Comment by alexanderkyte** [comment]
So it appears that on windows, ves_icall_System_Reflection_MonoMethodInfo_get_parameter_info is returning a type that goes into param_types for which the ```param_names [i] = param_types [i].ParameterType.AssemblyQualifiedName``` line in ```CADMessages.cs``` gets null. After looking at it, I think it might be related to GC Handles.

**Comment by alexanderkyte** [comment]
@monojenkins build Linux i386

**Comment by alexanderkyte** [comment]
Am I cleared to merge?

File: mcs/class/corlib/System.Runtime.Remoting.Messaging/CADMessages.cs [hunk 1]
```diff
@@ -97,9 +97,39 @@ public MethodBase Resolve ()
 			else
 				sig_cand = type.GetMethod (methodName, BindingFlags.Public|BindingFlags.NonPublic|BindingFlags.Instance, null, param_types, null);
 
+			if (sig_cand != null && generic_arg_names != null && !sig_cand.IsGenericMethodDefinition)
```
Comment (vargaz) [inline mcs/class/corlib/System.Runtime.Remoting.Messaging/CADMessages.cs]: Why is this affected by dedup which should have no visible effect on execution ?

**Comment by vargaz** [comment]
Otherwise looks ok.

File: mcs/class/corlib/System.Runtime.Remoting.Messaging/CADMessages.cs (hunk 1 above)
Comment (alexanderkyte) [reply mcs/class/corlib/System.Runtime.Remoting.Messaging/CADMessages.cs]: The mkbundle PR moves the test dlls out of the current directory into a subdirectory of the profile dir in lib. When the test is run out of the mcs/class subdir, the dll is then not in the current working directory. On windows, assemblies in the current working directory end up loaded in running domains, so the cross-appdomain call code uses the fast path that doesn't use the CADMessage remoting code. By moving the .dll, we made that dead code run on Windows. This dead code had a bug when the method is inflated, and one of the parameters ends up in the argument. That argument gets marshaled by type name, and the AssemblyQualifiedName of that MVAR returns null, putting a null in that array. When we later try to fetch the type and pass a null, it throws.

**Pull request merged**